dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/IntervalExpander.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector;
+
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+
+/**
+ * Expands a given interval based on conditions. MUST engulf the original interval i.e. -
```

Comment:
nit: I would suggest using `enclose` instead of `engulf`.

https://guava.dev/releases/32.1.3-jre/api/docs/com/google/common/collect/Range.html#encloses(com.google.common.collect.Range)